--- ../src-base/minecraft/net/minecraft/item/Item.java
+++ ../src-work/minecraft/net/minecraft/item/Item.java
@@ -635,7 +635,7 @@
      */
     public int getEntityLifespan(ItemStack itemStack, World world)
     {
-        return 6000;
+        return world.spigotConfig.itemDespawnRate;
     }
 
     /**
@@ -925,7 +925,7 @@
     {
         stack.itemDamage = damage;
 
-        if (stack.itemDamage < 0)
+        if (stack.itemDamage < 0 && stack.item != Items.MAP)
         {
             stack.itemDamage = 0;
         }
